### PR TITLE
tf: shadow t var to avoid using wrong one

### PR DESCRIPTION
**Pull request by howardjohn**
Its incredibly common in our tests to see:
```go
ctx.NewTest().Run(ctx, func() {
   t.Fatal()
}
```

This causes go to panic. They should instead be using ctx.Fatal.

Fixing and preventing all of these cases is hard. Instead, we can just
shadow t so its impossible to make this mistake. I strongly suspect
every new test is copy+pasting previous ones, so if we change all usages
of `ctx` to `t` I doubt more usages will pop up.

I did this only in one tests for now to get early feedback

### CODE REVIEW

File: tests/integration/pilot/analyze_test.go [hunk 1]
```diff
@@ -119,15 +120,15 @@ func TestDirectoryWithRecursion(t *testing.T) {
...
 			g := NewWithT(t)
 
-			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+			ns := namespace.NewOrFail(t, t, namespace.Config{
 				Prefix: "istioctl-analyze",
 				Inject: true,
 			})
 
-			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+			istioCtl := istioctl.NewOrFail(t, t, istioctl.Config{})
```
Comment (howardjohn) [inline tests/integration/pilot/analyze_test.go:131]: we may just make this only take in a resource.Context potentially. Probably a future PR

**Review by stevenctl — COMMENTED** [review]
There area. ton of improper `*testing.T` uses in the tests, mostly security and telemetry though.

**Review by stevenctl — APPROVED** [review]

**Comment by howardjohn** [comment]
@nmittler @Monkeyanator any opinions before I go change 1000 references?

**Review by Monkeyanator — APPROVED** [review]
LGTM, thanks for taking this on.

**Comment by jacob-delgado** [comment]
LGTM. The project long ago decided to not lint for shadowed variables. Fixing tests so they exit gracefully is good to make sure everything gets logged correctly.

**Review by jacob-delgado — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

**Pull request merged**